onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -240,10 +240,37 @@ def aten_any(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_arange(end: float) -> TensorType:
+@torch_op("aten::arange")
+def aten_arange(end: TReal, dtype: int = -1) -> TReal:
     # arange(Scalar end, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
 
-    raise NotImplementedError()
+    if dtype != -1:
+        end = op.Cast(end, to=dtype)
+
+    return op.Range(0, end, 1)
+
+
+@torch_op("aten::arange", overload=True)
+def aten_arange_start(start: TReal, end: TReal, dtype: int = -1) -> TReal:
```

Comment:
Since TReal has limited dtypes, what if the given dtype doesn't exist in the scope of TReal?